doc/programming.rst

```diff
@@ -146,6 +146,73 @@ The double quotes enclosing the sentence make the program recognize that
 the following characters are just that, *i.e.*, a sequence of characters
 (called a string) and not programming directives or variables.
 
+The Fortran package manager
+---------------------------
+
+The Fortran package manager (fpm) is a recent advancement introduced by the
+Fortran community. It helps to abstract common tasks when developing Fortran
+applications like building and running executables or creating new projects.
+
+.. warning::
+
+   The Fortran package manager is still a relatively new project.
+   Using new and recent software always has some risks of running into
+   unexpected issues. We have carefully evaluated fpm and the advantage
+   of the simple user interface outweigh potential issues.
+
+   This course is structured such that it can be completed with or without using fpm.
+
+.. note::
+
+   If you are doing this course on a machine in the Mulliken center you
+   have to activate the fpm installation by using the ``module`` commands:
+
+   .. code:: shell
+
+      module use /home/abt-grimme/modulefiles
+      module load fpm
+
+Create a new project with fpm by
+
+.. code-block:: none
+
+   fpm new --app myprogram
+
+This will initialize a new Fortran project with a simple program setup.
+Enter the newly created directory and run it with
+
+.. code-block:: none
+
+   cd myprogram
+   fpm run
+   ...
+    hello from project myprogram
+
+You can inspect the scaffold generated in ``app/main.f90`` and find
+a similar program unit as in your very first Fortran program:
+
+.. literalinclude:: src/fpm.1.f90
+   :language: fortran
+   :caption: app/main.f90
+   :linenos:
+
+Modify the source code with your editor of choice and simply invoke fpm run again.
+You will find that fpm takes care of automatically rebuilding
+your program before running it.
+
+.. tip::
+
+   By default fpm enables compile time checks and run time checks that
+   are absent in the plain compiler invocation. Those can help you catch
+   and avoid common errors when developing in Fortran.
+
+   To read more on the capabilities of fpm check the help page with
+
+   .. code-block::
+
+      fpm help
+
```

Comment:
Maybe it would be beneficial for the students to use fpm build instead of fpm run?

Reply:
But is it possible to use fpm run with a command-line syntax? I'm not sure, how this would work?

Reply:
Oh, you can pass-through arguments by the usual Unix like shell syntax:
```
fpm run -- --arg
```
Everything after ``--`` goes unmodified as argument to the executable.

Reply:
Oh, great - I did not know that

Reply:
But you are right, I should highlight this feature since it will be used in the SCF part.